Database/MySQL/PreparedStatement.hs

```diff
@@ -0,0 +1,485 @@
+            k defaultBind
+        TinyInt i ->
+            with i $ \ptr ->
+                k $ defaultBind
+                    { mysqlBindBufferType = Types.Tiny
+                    , mysqlBindBuffer = castPtr ptr
+                    }
+        Short i ->
+            with i $ \ptr ->
+                k $ defaultBind
+                    { mysqlBindBufferType = Types.Short
+                    , mysqlBindBuffer = castPtr ptr
+                    }
+
+        MediumInt i ->
+            with i $ \ptr ->
+                k $ defaultBind
+                    { mysqlBindBufferType = Types.Long
+                    , mysqlBindBuffer = castPtr ptr
+                    }
+
+        Long i ->
+            with i $ \ptr ->
+                k $ defaultBind
+                    { mysqlBindBufferType = Types.Long
+                    , mysqlBindBuffer = castPtr ptr
+                    }
+        LongLong i ->
+            with i $ \ptr ->
+                k $ defaultBind
+                    { mysqlBindBufferType = Types.LongLong
+                    , mysqlBindBuffer = castPtr ptr
+                    }
+        UnsignedTinyInt i ->
+            with i $ \ptr ->
+                k $ defaultBind
+                    { mysqlBindBufferType = Types.Tiny
+                    , mysqlBindBuffer = castPtr ptr
+                    , mysqlBindIsUnsigned = 1
+                    }
+        UnsignedShort i ->
+            with i $ \ptr ->
+                k $ defaultBind
+                    { mysqlBindBufferType = Types.Short
+                    , mysqlBindBuffer = castPtr ptr
+                    , mysqlBindIsUnsigned = 1
+                    }
+
+        UnsignedMediumInt i ->
+            with i $ \ptr ->
+                k $ defaultBind
+                    { mysqlBindBufferType = Types.Long
+                    , mysqlBindBuffer = castPtr ptr
+                    , mysqlBindIsUnsigned = 1
+                    }
+
+        UnsignedLong i ->
+            with i $ \ptr ->
+                k $ defaultBind
+                    { mysqlBindBufferType = Types.Long
+                    , mysqlBindBuffer = castPtr ptr
+                    , mysqlBindIsUnsigned = 1
+                    }
+        UnsignedLongLong i ->
+            with i $ \ptr ->
+                k $ defaultBind
+                    { mysqlBindBufferType = Types.LongLong
+                    , mysqlBindBuffer = castPtr ptr
+                    , mysqlBindIsUnsigned = 1
+                    }
+        Float f ->
+            with f $ \ptr ->
+                k $ defaultBind
+                    { mysqlBindBufferType = Types.Float
+                    , mysqlBindBuffer = castPtr ptr
+                    }
+        Double d ->
+            with d $ \ptr ->
+                k $ defaultBind
+                    { mysqlBindBufferType = Types.Double
+                    , mysqlBindBuffer = castPtr ptr
+                    }
+
+        Decimal _ ->
+            error "Unsupported input type: Decimal"
```

Comment:
Should we just treat this like a blob on input instead of throwing a runtime error?